desktop/cmp/input/DateInput.js

```diff
@@ -172,22 +179,25 @@ export class DateInput extends HoistInput {
     }
 
     renderButtons(enableClear, enablePicker) {
-        if (!enableClear && !enablePicker) return null;
+        const props = this.getNonLayoutProps(),
+            enableTextInput = withDefault(props.enableTextInput, true),
+            isClearable = (this.internalValue !== null),
+            isPickerOnlyMode = !enableTextInput && !props.disabled;
 
         return buttonGroup({
             padding: 0,
             items: [
                 button({
-                    omit: !enableClear,
+                    omit: !enableClear || !isClearable,
                     icon: Icon.cross(),
                     tabIndex: -1, // Prevent focus on tab
                     onClick: this.onClearBtnClick
                 }),
                 button({
-                    omit: !enablePicker,
+                    className: enablePicker ? '' : 'disabled-picker-icon',
                     icon: Icon.calendar(),
-                    tabIndex: -1, // Prevent focus on tab
-                    onClick: this.onPopoverBtnClick
+                    tabIndex: isPickerOnlyMode ? undefined : -1, // Prevent focus on tab, unless in pickerOnly mode
```

Comment:
This fixes our focus on tab problem without having to move the picker button out of the disabled input. A little embarrassed at how long this took me to figure out given the incredibly obvious comment that I apparently just didn't read about 100 times I looked at this code. :/